fs/io_zenfs.cc

```diff
@@ -547,32 +633,30 @@ IOStatus ZonedWritableFile::Flush(const IOOptions& /*options*/,
 }
 
 IOStatus ZonedWritableFile::RangeSync(uint64_t offset, uint64_t nbytes,
-                                      const IOOptions& options,
-                                      IODebugContext* dbg) {
-  if (wp < offset + nbytes) return Fsync(options, dbg);
+                                      const IOOptions& /*options*/,
+                                      IODebugContext* /*dbg*/) {
+  if (wp < offset + nbytes) return DataSync();
 
   return IOStatus::OK();
 }
 
-IOStatus ZonedWritableFile::Close(const IOOptions& options,
-                                  IODebugContext* dbg) {
-  Fsync(options, dbg);
-  return zoneFile_->CloseWR();
+IOStatus ZonedWritableFile::Close(const IOOptions& /*options*/,
+                                  IODebugContext* /*dbg*/) {
+  IOStatus s = DataSync();
+  if (!s.ok()) return s;
+
+  s = zoneFile_->CloseWR();
+  if (!s.ok()) return s;
+
+  return zoneFile_->PersistMetadata();
```

Comment:
Do we need this change? Seems like it does the same as calling `Fsync` but with more lines of code.

Reply:
It is needed, as we need to do a CloseWR() before PersistMetadata()

Reply:
Why is sync->persist->closeWR not good ?

Reply:
We need to set extent_start_ = NO_EXTENT before persisting metadata